packages/contractkit/src/identity/offchain-data-wrapper.ts

```diff
@@ -57,10 +60,11 @@ export default class OffchainDataWrapper {
     this.signer = signer || self
   }
 
-  async readDataFromAsResult(
-    account: string,
-    buildTypedData: (x: Buffer) => Promise<EIP712TypedData>,
-    dataPath: string
+  async readDataFromAsResult<DataType>(
+    account: Address,
+    dataPath: string,
+    checkOffchainSigners: boolean,
+    type?: t.Type<DataType>
```

Comment:
Minor suggestion: I imagine the `checkOffchainSigners` will almost always be `true`, so moving it to the last position and providing a default will make caller code slightly more concise.
```suggestion
    type?: t.Type<DataType>,
    checkOffchainSigners: boolean = true,
```

Reply:
That doesn't work because you then need to pass a type, this API isn't great I'll give you that. The absence of a type means it's binary data and therefore io-ts cannot help.

With your proposed changes, to pass `checkOffchainSigners` you'd need to pass a type